composition-js/src/__tests__/compose.demandControl.test.ts

```diff
@@ -0,0 +1,552 @@
+      };
+  
+      const result = composeServices([subgraphA, subgraphB]);
+      assertCompositionSuccess(result);
+      expect(result.hints).toMatchInlineSnapshot(`
+        Array [
+          CompositionHint {
+            "coordinate": undefined,
+            "definition": Object {
+              "code": "MERGED_NON_REPEATABLE_DIRECTIVE_ARGUMENTS",
+              "description": "A non-repeatable directive has been applied to a schema element in different subgraphs with different arguments and the arguments values were merged using the directive configured strategies.",
+              "level": Object {
+                "name": "INFO",
+                "value": 40,
+              },
+            },
+            "element": undefined,
+            "message": "Directive @cost is applied to \\"Query.sharedWithCost\\" in multiple subgraphs with different arguments. Merging strategies used by arguments: { \\"weight\\": MAX }",
+            "nodes": undefined,
+          },
+        ]
+      `);
+    });
+  
+    it('hints about merged @listSize arguments', () => {
+      const subgraphA = {
+        name: 'subgraph-a',
+        typeDefs: asFed2SubgraphDocument(gql`
+          extend schema @link(url: "https://specs.apollo.dev/listSize/v0.1", import: ["@listSize"])
+  
+          type Query {
+            sharedWithListSize: [Int] @shareable @listSize(assumedSize: 10)
+          }
+        `)
+      };
+      const subgraphB = {
+        name: 'subgraph-b',
+        typeDefs: asFed2SubgraphDocument(gql`
+          extend schema @link(url: "https://specs.apollo.dev/listSize/v0.1", import: ["@listSize"])
+  
+          type Query {
+            sharedWithListSize: [Int] @shareable @listSize(assumedSize: 20)
+          }
+        `)
+      };
+  
+      const result = composeServices([subgraphA, subgraphB]);
+      assertCompositionSuccess(result);
+      expect(result.hints).toMatchInlineSnapshot(`
+        Array [
+          CompositionHint {
+            "coordinate": undefined,
+            "definition": Object {
+              "code": "MERGED_NON_REPEATABLE_DIRECTIVE_ARGUMENTS",
+              "description": "A non-repeatable directive has been applied to a schema element in different subgraphs with different arguments and the arguments values were merged using the directive configured strategies.",
+              "level": Object {
+                "name": "INFO",
+                "value": 40,
+              },
+            },
+            "element": undefined,
+            "message": "Directive @listSize is applied to \\"Query.sharedWithListSize\\" in multiple subgraphs with different arguments. Merging strategies used by arguments: { \\"assumedSize\\": NULLABLE_MAX, \\"slicingArguments\\": NULLABLE_UNION, \\"sizedFields\\": NULLABLE_UNION, \\"requireOneSlicingArgument\\": NULLABLE_AND }",
+            "nodes": undefined,
+          },
+        ]
+      `);
+    });
+  });
+});
+
+describe('demand control directive extraction', () => {
```

Comment:
I would put these tests into their own file extractSubgraphsFromSupergraph.demandControl.test.ts

Reply:
The main reason I didn't do this is because I wanted to share the subgraph templates between the composition and extraction tests. The existing extraction tests are in internals-js, so importing them from here would introduce a circular dependency in those packages.